src/services/ethereum/CustomSigner.js

```diff
@@ -57,7 +57,7 @@ export default function CustomSigner(privateKey: string, provider: string, app:
             amount: transactionObject.value,
             estimate: estimate.toString(),
             app,
-            gasLimit: transactionObject.gasLimit,
+            gasLimit: estimate.toString(), // get gas limit from ether.js service https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_estimategas
```

Comment:
This one looks like a solution, but I'm not sure if it's a correct one. Let me explain: gasLimit is a value that specify an upper bound for gas used during transaction. Before transaction is done, it's checked that wallet have enough funds to cover `gasLimit*gasPrice + transactionValue`, so that even if it takes gasLimit, requester has funds to pay for computation.
Now it's clear where the problem was - user was shown 'insufficient funds' error because even though estimated transaction amount is less than user's balance, it doesn't have enough funds to cover `gasLimit*gasPrice + transactionValue`.

So, provided solution helps in cases when `estimate` is less than `gasLimit` and therefore we can set `estimate` as limit, because we sure we don't need more (at least with high probability). However, if `estimate` is huge compared to `gasLimit`, then we likely should use provided `gasLimit` anyway, because it could be set intentionally to cut off amount of maximal gas used. 

Proposed solution: use minimal from `estimate` and provided `gasLimit` (if one is specified). 
Be sure to use operations on BigNumber.

Reply:
According to the comment. This is pulled from eth_estimategas via the ethereum API. The estimate is a conservative estimate, usually returning a safely high number that can be used for gas limit.

May want to check/cite the api documentation to be sure

Reply:
I think I misunderstood @seland's comment, so my comment may not be relevant. I'll get out of the way now :)

Reply:
my understand that transactionObject.gasLimit is always set to DEFAULT_GAS_LIMIT=21000, and we need some way to calculate the right gasLimit so the transaction could be fired without insufficient fund error. As I research that we can use ethers.js util with the gasEstimate function for doing that. pls check https://github.com/ethers-io/ethers.js/issues/194